### DISCUSSION TITLE
PowerOnState and Relay with pulse opening door

### DISCUSSION THREAD

**Discussion by mtraxiot** [arendst/Tasmota#12619] [Q&A] [answered]
I have two Relays connected to my Garage Door controlling them using a PULSE 7 ON/OFF switch, this works fine Except when the device gets a power cycle which seems to sends a Pulse to both doors ( ie opens them), This is not ideal.

I've set poweronstate to OFF 
PowerOnState 0
SetOption0 0
Both Reed switches are set to MQTT messages only
switchmode1 15
switchmode2 15

Am I missing anything else?

**Comment by joba-1** [comment, marked as answer]
Some pins of esp or your board have hardware defined levels (pullup/pulldown). Test other pins. You probably want one with a pulldown.

**Reply by sfromis** [reply]
What may be more relevant than pullup/down is that a few pins (gpio 1,2,3,16) are well-known to have a high pulse at boot, due to the hardware design of the ESP8266.

**Reply by mtraxiot** [reply]
Thanks however I've put a extra pull up resistor on both pins, and I did try another adjacent pins but same thing occurred.  Note I'm using a esp32 chip.

**Reply by sfromis** [reply]
It still matters a lot which pins you use. On ESP32, expect boot activity on gpio 0,1,3,5,14,15, at least.

**Comment by joba-1** [comment]
… and pullup is bad: While off, pin is low, if you switch on, while esp32 still has not initialized pins, the pullup sends high to your door. Then esp does its thing (see sfromis) and then tasmota does what you told it -> set pin low. Switch to open door complete.

**Reply by mtraxiot** [reply]
Correction to my last post I'm using pins 13 and 27 with no extra resistor.

**Reply by mtraxiot** [reply]
I've used a connected relay on a NodeMCU ESP8266 chip for a few years without a pull up/down resister,  
why do I need one with TASMOTA and ESP32?

**Reply by sfromis** [reply]
No one suggested that you **need** to add a pull up/down resistor. But in case of an existing pull resistor on the board, that could conflict with what you try to do. And if what you try to do is not electrically compatible, a pull up/down resistor may hide a more basic problem. 

You are avoiding specific details, but one example of "not electrically compatible" could be a setup expecting active low, which could be triggered during power up, before Tasmota can activate the pin.

Another potential issue (which may react differently at boot time) is if your relay is specified for 5V signal input, but only gets 3.3V, or tries to draw more power than available. As ESP32 needs extra power when booting and connecting, insufficient available power could also cause trouble.

**Reply by mtraxiot** [reply]
I think what's happening is there appears to be a issue with power requirements during boot is somehow causing the problem. I'm having the same problem with mongoose os firmware,  but I eventually solved it by delaying the initialisation of the relay pins till most of rest of the OS had finished.
so is that possible with tasmota?

**Reply by sfromis** [reply]
You have been awfully silent about which pins. Also expect boot activity on GPIO 16 and 17.

**Reply by mtraxiot** [reply]
I'm using pin  gpio 27
